OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -36,29 +38,44 @@ public Fly(Actor self, Target t, WDist minRange, WDist maxRange)
 			this.minRange = minRange;
 		}
 
-		public static void FlyToward(Actor self, Aircraft plane, int desiredFacing, WDist desiredAltitude)
+		public static void FlyToward(Actor self, Aircraft aircraft, int desiredFacing, WDist desiredAltitude)
 		{
-			desiredAltitude = new WDist(plane.CenterPosition.Z) + desiredAltitude - self.World.Map.DistanceAboveTerrain(plane.CenterPosition);
+			desiredAltitude = new WDist(aircraft.CenterPosition.Z) + desiredAltitude - self.World.Map.DistanceAboveTerrain(aircraft.CenterPosition);
 
-			var move = plane.FlyStep(plane.Facing);
-			var altitude = plane.CenterPosition.Z;
+			var move = aircraft.FlyStep(aircraft.Facing);
+			var altitude = aircraft.CenterPosition.Z;
 
-			plane.Facing = Util.TickFacing(plane.Facing, desiredFacing, plane.TurnSpeed);
+			aircraft.Facing = Util.TickFacing(aircraft.Facing, desiredFacing, aircraft.TurnSpeed);
 
 			if (altitude != desiredAltitude.Length)
 			{
-				var delta = move.HorizontalLength * plane.Info.MaximumPitch.Tan() / 1024;
+				var delta = move.HorizontalLength * aircraft.Info.MaximumPitch.Tan() / 1024;
 				var dz = (desiredAltitude.Length - altitude).Clamp(-delta, delta);
 				move += new WVec(0, 0, dz);
 			}
 
-			plane.SetPosition(self, plane.CenterPosition + move);
+			aircraft.SetPosition(self, aircraft.CenterPosition + move);
+		}
+
+		public static bool AdjustAltitude(Actor self, Aircraft aircraft, WDist targetAltitude)
+		{
+			targetAltitude = new WDist(aircraft.CenterPosition.Z) + targetAltitude - self.World.Map.DistanceAboveTerrain(aircraft.CenterPosition);
```

Comment:
This is converting an altitude (relative to the ground) to an absolute Z offset.  It would be cleaner if we could change this to `var targetZ` or `var absoluteAltitude` or another name that makes it clear and self-documenting (and add a comment if needed).